misc/systemd/ipfs.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
+description=ipfs p2p daemon
+After=network.target
+Requires=network.target
+
+[Service]
+Type=simple
+User=ipfs
+RestartSec=1
+Restart=always
+PermissionsStartOnly=true
+Nice=18
+StateDirectory=/var/lib/ipfs
```

Comment:
I don't have an issue with `/var/lib/ipfs` being the place that the repo gets stored but considering the default `StorageMax` of 4TB, which is too large for a general purpose service file, but also most suggested partitioning schemes for linux distros when installing a new distro, places more storage towards the `/home` directory instead of the `/` directory.

Reply:
The administrator of a service (or container) can mount storage wherever they'd like.  This stores state data in /var/lib like every other service stores state data in /var/lib: /var/lib/httpd, /var/lib/ldap, /var/lib/mysql, et c.  If you have a problem with this you have a problem with LSB.

Reply:
So, this isn't quite the correct usage of `StateDirectory`. It's primarily used for the `DynamicUser` feature and should be relative (that is, just `ipfs`).

So, we should either drop this *or* switch this unit to a dynamic user. If we go with the latter, we'd want to:

1. Not have a tmpfiles/sysusers config file and *not* create a user account.
2. Set `DynamicUser=true`
3. Set `StateDirectory=ipfs`

The *issue* with this is that, if there's a UID conflict, systemd will have to `chown` the entire directory tree (which, depending on the datastore, could be very expensive).

On the other hand, it means one config file versus 3.